test/unit/fact_cleaner_test.rb

```diff
@@ -50,11 +51,8 @@ class FactCleanerTest < ActiveSupport::TestCase
   end
 
   test "it cleans excluded facts" do
-    name_generator_type = Setting[:name_generator_type]
-    host_owner = Setting[:host_owner]
-    Setting.stubs(:[]).with(:excluded_facts).returns(['ignored*', '*bad'])
-    Setting.stubs(:[]).with(:name_generator_type).returns(name_generator_type)
-    Setting.stubs(:[]).with(:host_owner).returns(host_owner)
+    Setting.stubs(:cache).returns(null_store)
```

Comment:
Shouldn't this rather be done globally on the test environment preparation in the test helper? Don't we otherwise risk, that the stubbed method would potentially influence other tests and it would be order dependent, as it seems that was the issue before the patch?

Reply:
It should, but let's please do it in a separate PR and not on a "refs #" PR. :-)

Reply:
I have a proposal:

1. lets try to do the stubbing globally in test helper
2. if everything still works, let's keep it
3. if there are some failures discovered that would be independent of the original issue, let's keep this version.

Deal?

Reply:
Fair enough, deal.

Reply:
@timogoebel I know you'll probably hate me  by now, but since I was not able to run the katello tests successfully (with or without the patch, I tried 1.16 + katello 3.5), and since I had a wrong assumption to begin with, for now, I would suggest getting back to the version, where the stubbing of `:cache` was done in the `fact_cleaner_test`, to minimize the potential negative impact on others. Sorry again :/

Reply:
@iNecas: Don't worry, I'll restore the original version. Luckily I have a backup.